subsys/bluetooth/host/conn.c

```diff
@@ -2671,19 +2671,21 @@ int bt_conn_le_create(const bt_addr_le_t *peer,
 	create_param_setup(create_param);
 
 #if defined(CONFIG_BT_SMP)
-	if (!bt_dev.le.rl_size || bt_dev.le.rl_entries > bt_dev.le.rl_size) {
-		bt_conn_set_state(conn, BT_CONN_CONNECTING_SCAN);
+	if (IS_ENABLED(CONFIG_BT_PRIVACY)) {
+		if (!bt_dev.le.rl_size || bt_dev.le.rl_entries > bt_dev.le.rl_size) {
```

Comment:
Minor, but you could also have done
```suggestion
	if (IS_ENABLED(CONFIG_BT_PRIVACY) &&
	    !bt_dev.le.rl_size || bt_dev.le.rl_entries > bt_dev.le.rl_size) {
```
to avoid indententing the code below

Reply:
Tip: You can hide white-space changes. It's very nice with split view.
![image](https://user-images.githubusercontent.com/48479610/178224488-3cb3e4d2-f13c-4568-9433-bc998d74be43.png)

Reply:
Use `&&` instead of added indentation, and redundant `if` clause.